tuplex/test/core/LoopTest.cc

```diff
@@ -1103,4 +1103,178 @@ TEST_F(LoopTest, CodegenTestLoopInIf) {
+                "        x += a * c\n"
+                "        y += b\n"
+                "    return (x, y)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0], Row(50, "ab"));
+}
+
+TEST_F(LoopTest, CodegenTestLoopWithIteratorGeneralI) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    t = iter([1, 2, 3, 4])\n"
+                "    count = 0\n"
+                "    for a, b in zip(t, t):\n"
+                "        count += 1\n"
+                "        x += a + b\n"
+                "    return (count, x)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0], Row(2, 10));
+}
+
+TEST_F(LoopTest, CodegenTestLoopWithIteratorGeneralII) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    t = iter([1, 2, 3, 4])\n"
+                "    q = iter('ab')\n"
+                "    p = enumerate((10, 20, 30, 40))\n"
+                "    for a, b, c in zip(t, q, p):\n"
+                "        continue\n"
+                "    r1 = next(t)\n"
+                "    r2 = next(p)\n"
+                "    return (r1, r2)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0], Row(4, Tuple(2, 30)));
+}
+
+TEST_F(LoopTest, CodegenTestLoopWithIteratorGeneralIII) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    l = iter([1000.5, 2000.5, 3000.0, 4000.0, 5000.5, x])\n"
+                "    for index, value in enumerate(l):\n"
+                "        if index == 3:\n"
+                "            break\n"
+                "    b1 = next(l)\n"
+                "    b2 = next(l)\n"
+                "    return (b1, b2)";
+
+    auto v = c.parallelize({
+        Row(6000.0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0], Row(5000.5, 6000.0));
+}
+
+TEST_F(LoopTest, CodegenTestLoopWithIteratorGeneralIV) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    a = 'a!b!c!d!A!B!C!D!'\n"
+                "    t = enumerate(a, 4)\n"
+                "    for i, v in t:\n"
+                "        if i == 5:\n"
+                "            b1 = v\n"
+                "        if i == 10:\n"
+                "            b2 = v\n"
+                "        if i == 16:\n"
+                "            break\n"
+                "    else:\n"
+                "        next(t)\n"
+                "    c = next(t)\n"
+                "    return (b1, b2, c)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
```

Comment:
Can we add tests where the the row type has. acolumn of list type and then some for loop over that data is executed?

E.g. computing mean/std of the passed lists per row?

(10, [1, 2, 3, 4])
(20, [2, 3, 4])

(Not sure, was this done maybe already?)

Reply:
Added tests.